job/dependency_resolver.go

```diff
@@ -156,6 +156,11 @@ func (r *dependencyResolver) resolveStaticDependencies(ctx context.Context, jobS
 					}
 					projectName := depParts[0]
 					jobName := depParts[1]
+
```

Comment:
this change is not easy to understand, rather lets reconsider entire dependency resolution flow where
1. We move away from user specifiying inter/intra type & the job dependency is inferred automatically.
2. Lets change the scope of this issue or lets consider in the new updated flow of Job Deploy enhancement

Reply:
If we want to apply the point (1) you mentioned, then in my opinion, we have need to do the point (2). If not, more conflicts could happend.